### PR TITLE
Add block separators to list category

### LINKED ISSUE #1645: List block palette is missing separators
### Expected Behavior

There should be separators between several of the blocks in the List category: between "add to" and "delete of"; between "replace item" and "item of"; and between "list contains" and "show list". See the screenshot of 2.0 below:

Screenshot of list category in 2.0, with the aforementioned separators

### Actual Behavior

There are no separators in the list block palette. See the 3.0 screenshot:

Screenshot of list category in 3.0, without any separators

### Steps to Reproduce

Create a list in the "Variables" (or "Data") category.

### PR DESCRIPTION
**Pull request by kyleplo**
### Resolves

Resolves #1645

### Proposed Changes

Adds a method to add separators to the category and places the method in the appropriate places.

### Reason for Changes

Scratch 2.0 has separators in the list category.

### CODE REVIEW

File: core/data_category.js [hunk 1]
```diff
@@ -472,3 +475,16 @@ Blockly.DataCategory.createValue = function(valueName, type, value) {
       '</value>';
   return valueField;
 };
+/**
```
Comment (towerofnix) [inline core/data_category.js]: Code cleanup - there should be a line break before this one.

Comment (kyleplo) [reply core/data_category.js]: Done!

File: core/data_category.js [hunk 2]
```diff
@@ -472,3 +475,17 @@ Blockly.DataCategory.createValue = function(valueName, type, value) {
...
+ * @param {!Array.<!Element>} xmlList Array of XML block elements.
+ */
+Blockly.DataCategory.addSep = function(xmlList) {
+    var gap = 36;
+    var sepText = '<xml>' +
+        '<sep gap="' + gap + '"/>' +
+        '</xml>';
+    var sep = Blockly.Xml.textToDom(sepText).firstChild;
+    xmlList.push(sep);
+  }
```
Comment (rachel-fenichel) [inline core/data_category.js]: You've got an extra close-brace here that's causing travis failures.

Comment (kyleplo) [reply core/data_category.js]: I tried, and it didn't work.

**Comment by rachel-fenichel** [comment]
@kchadha code lgtm, assuming that you want this.

File: core/data_category.js (hunk 2 above)
Comment (kyleplo) [reply core/data_category.js]: Oh, now it's working.

**Review by kchadha — APPROVED** [review]
Looks good! Thanks for adding this, @kyleplo.

**Pull request merged**